### PR TITLE
Dotty support

### PR DESCRIPTION
**Pull request by julienrf**
This PR includes work from @odersky and @biboudis as well as additional stuff to support cross compilation for both Scala 2.12 and Dotty.

I still have execution errors when running the tests:

~~~
Test strawman.collection.test.StrawmanTest.mainTest failed: java.lang.IncompatibleClassChangeError: Conflicting default methods: strawman/collection/SeqLike.coll strawman/collection/Iterable.coll
~~~

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -69,12 +49,24 @@ val collections =
           password <- sys.env.get("SONATYPE_PASSWORD")
         } yield Credentials("Sonatype Nexus Repository Manager", "oss.sonatype.org", username, password)
       ).toList
-    ).configure(dottyEnable)
+    )
+    // Dotty support
+    .settings(
+      libraryDependencies ++= {
+        if (targetingDotty.value) Seq("ch.epfl.lamp" % "dotty_2.11" % scalaVersion.value % "scala-tool")
```
Comment (smarter) [inline build.sbt]: This shouldn't be necessary with sbt 0.13.15

Comment (julienrf) [inline build.sbt]: Thanks I removed it!

**Comment by julienrf** [comment]
I updated the build according to @smarter review. Has anyone an idea of what does [this error](https://travis-ci.org/scala/collection-strawman/builds/223098835#L2420) mean?

**Comment by smarter** [comment]
Probably a bug in dotty

**Pull request merged**